forge-core/src/main/java/forge/card/CardType.java

```diff
@@ -788,6 +787,18 @@ public boolean sharesCardTypeWith(final CardTypeView ctOther) {
         return false;
     }
 
+    public boolean sharesAllCardTypesWith(final CardTypeView ctOther) {
+        if (ctOther == null) {
+            return false;
+        }
+        for (final CoreType type : getCoreTypes()) {
+            if (!ctOther.hasType(type)) {
```

Comment:
the script probably works already since you're checking both cards anyway

but the logic here should still be fixed, right now your approach is not symmetrical (imo comparing sizes will be cleanest)

Reply:
Would checking the sizes of each actually ensure they both have the same core types? I'm a bit confused on that.

Reply:
Hmn why not?
It's a Set so no worrying about any doubling

Reply:
The Core Types of one of the cards could be "Instant" and the other "Creature". Both Sets would have a size of 1 but the cards would not share types. Unless I am misunderstanding the rules of the card.

Reply:
haha no, you're absolutely right and I'm too sleepy already :p

anyway the problem with your current implementation is you might not catch if `ctOther` has any additional types as long as it has at least the ones from this instance

Reply:
I'll get that fixed up! Thank you :)